scripts/00_prepare.sh

```diff
@@ -1,15 +1,12 @@
-#!/bin/bash -e
-# vim: set ts=4 sw=4 sts=4 et :
-### 00_prepare.sh : Download and extract the archlinux bootstrap mini-distro
+#! /bin/bash
+
 echo "--> Archlinux 00_prepare.sh"
 
 ARCHLINUX_PLUGIN_DIR="${ARCHLINUX_PLUGIN_DIR:-"${SCRIPTSDIR}/.."}"
-
 ARCHLINUX_SRC_PREFIX="${ARCHLINUX_SRC_PREFIX:-http://mirrors.kernel.org/archlinux}"
-ARCHLINUX_REL_VERSION="${ARCHLINUX_REL_VERSION:-$(date --date=yesterday '+%Y.%m').01}"
-BOOTSTRAP_TARBALL="archlinux-bootstrap-${ARCHLINUX_REL_VERSION}-x86_64.tar.gz"
+BOOTSTRAP_TARBALL=$(wget -qO- "$ARCHLINUX_SRC_PREFIX"/iso/latest/sha1sums.txt | grep bootstrap | awk '{print $2}')
```

Comment:
This is dangerous thing to do, if someone manages to replace sha1sums.txt with some malicious content, then will be able to mess with this script. Obvious thing would be include `../../...` in path and try to choose arbitrary file, but probably more scenarios are possible.

I'd recommend using `grep -o "archlinux-bootstrap-[0-9.]*.tar.gz"` instead of the current grep + awk.

Reply:
Fixed it.